addons/portal/controllers/mail.py

```diff
@@ -117,11 +22,13 @@ def _portal_post_has_content(self, thread_model, thread_id, message, attachment_
 
     @http.route('/mail/avatar/mail.message/<int:res_id>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
     def portal_avatar(self, res_id=None, height=50, width=50, access_token=None, _hash=None, pid=None):
-        """ Get the avatar image in the chatter of the portal """
+        """Get the avatar image in the chatter of the portal"""
         if access_token or (_hash and pid):
-            message = request.env['mail.message'].browse(int(res_id)).exists().filtered(
-                lambda msg: _check_special_access(msg.model, msg.res_id, access_token, _hash, pid and int(pid))
-            ).sudo()
+            message = request.env["mail.message"].browse(int(res_id)).exists().filtered(
+                lambda msg: request.env[msg.model]._get_thread_with_access(
+                    msg.res_id, token=access_token, hash=_hash, pid=pid and int(pid)
```

Comment:
`browse.exist()` here returns a record with access read.

Reply:
`exists` just makes sure the record exists in the database, but it shouldn't change ACL and in particular not add `sudo`.

When reading fields, standard ACL check should still apply, and if the route is called with token from a user with no ACL on the thread, the call should normally crash due to missing `sudo` when reading the fields (unless they are read somewhere else and already put it cache, but I don't see where it would be done).

Reply:
I see but public user has access read ([here](https://github.com/odoo/odoo/blob/a991ab8cf02e7a93bfd9285d049a8c85b0fa125d/addons/mail/security/ir.model.access.csv#L3)) to the message so I think it should be a record with read access.

Reply:
Message ACL is complex and mostly done with override of access check methods, and it is (among other things) related to access of related thread.

Reply:
So after investigation: `browse + field access` on message is technically equivalent to `browse sudo field access` because ACL are naive and neither `_search` nor `check_access_rule` overrides are called when reading a field.

That's a pretty bad limitation in terms of security but I guess that's fine if we know about it. It would be nice to have field fetching covered by overrides too though.

Reply:
``So after investigation: browse + field access on message is technically equivalent to browse sudo field access because ACL are naive and neither _search nor check_access_rule overrides are called when reading a field.``

Wait, what :D ?